Get Started: explaining code.zip contents, and plots axis options

xnutsive opened the pull request:
1. Added the more explicit `ls -R` to show that the code.zip already has params.yaml. 
2. Added `dvc plots modify` details to experiments guide.
 
---

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

jorgeorpinel commented on content/docs/start/data-pipelines.md:
````diff
@@ -26,11 +26,17 @@ Get the sample code like this:
+evaluate.py       featurization.py  prepare.py        requirements.txt  train.py
 ```
 
+`params.yaml` contains training and evaluation parameters, we'll cover them later in the guide. 
+
+`src` contains the python code we'll use in this guide. 
+
````
Showing params.yaml above seems good to me indeed, since the file is used later in this same page. But these lines explaining what the dir entries are is probably unnecessary, can you roll back (remove) these 4 lines?

shcheklein replied:
yeah, src is self-explanatory I think

at this stage we should probably replace `ls` with `tree` to show the better output

`params.yaml` - I would remove "we'll cover them later in the guide. " to avoid creating a feeling that it is something DVC-specific.

jorgeorpinel replied:
Agree about `tree` and not explaining src/

I would also try to avoid explaining params.yaml here since it's explained later but if needed, a short sentence as part of the paragraph below should suffice.

xnutsive replied:
Yeah, a sentence like "`params` contains params" doesn't really explain much. Let me omit that all together.

shcheklein commented on content/docs/start/experiments.md:
```diff
@@ -211,12 +211,257 @@ scores.json  auc       0.61314  0.07139
 
+<details>
+
+### 💡 Expand to see what plots modify does under the hood
```
some duplication happening in the text below?

jorgeorpinel replied:
Yes, something is wrong. It looks like over 200 lines were added and the `<details>` may not be closed properly.

jorgeorpinel replied:
But do we even want to add all this info about `plots modify`? See also https://github.com/iterative/dvc.org/pull/1514#pullrequestreview-441099599 below.

shcheklein replied:
yeah, it should be short and simple with a link to the CR for the details

jorgeorpinel replied:
p.s. @xnutsive you can sanity check in https://dvc-landing-xnutsive-ge-1visdu.herokuapp.com/doc/start/experiments. This is how it looks rn:

![image](https://user-images.githubusercontent.com/1477535/86279073-0cf81300-bb9f-11ea-9a77-233649be0c1a.png)

jorgeorpinel commented on content/docs/start/experiments.md:
````diff
@@ -211,12 +211,257 @@ scores.json  auc       0.61314  0.07139
 And finally, we can compare `ROC` curves with a single command!
 
 ```dvc
+$ dvc plots modify prc.json -x recall -y precision
+$ git commit -am "Update plots format"
+
 $ dvc plots diff
 file:///Users/dvc/example-get-started/plots.html
 ```
````
What's the point/need for this addition? This is no longer "a single command".

Would also need to update the image underneath. I think we want to keep this simple, no need for modify here.

shcheklein replied:
@jorgeorpinel it's needed indeed to get the image below :) May be we can change or move it somehow (e.g. hide into the expandable section).

jorgeorpinel replied:
Adding a full `dvc plots modify` to the guide just so the plot axis labels match the image presented doesn't seem super necessary to me. But OK, up to you. Will just need to change the "with a single command" text above, and fix the plots modify expandable section below (per https://github.com/iterative/dvc.org/pull/1514#pullrequestreview-441093000).

jorgeorpinel replied:
Oh I see it's not just the labels but the actual fields used as data sources. OK

jorgeorpinel replied:
Well, we could just add -x and -y to the plots diff command, actually. See https://dvc.org/doc/command-reference/plots/diff#options

jorgeorpinel commented on content/docs/start/experiments.md:
```diff
@@ -211,12 +211,257 @@ scores.json  auc       0.61314  0.07139
 
 ![](/img/plots_roc_get_started.svg)
 
+<details>
+
+### 💡 Expand to see what plots modify does under the hood
+
+[`dvc plots`](https://dvc.org/doc/command-reference/plots#plots) is a flexible visualisation tool for your continuous metrics. [`dvc plots modify`](https://dvc.org/doc/command-reference/plots/modify), in the example above, sets what to use as an X-axis what to plot on Y-axis of a visualisation when you run `dvc plots diff` later on.
```
Why does the GH diff here looks like you added over 200 lines to this doc? Something is wrong. Please double check the HTML and formatting.

jorgeorpinel replied:
See https://dvc.org/doc/user-guide/contributing/docs

jorgeorpinel replied:
I see this was already reported by Ivan. Moved to https://github.com/iterative/dvc.org/pull/1514#discussion_r448542869. Resolving here.

xnutsive commented:
Hey!

Thank you for investing so much of your attention to this. 🙌

I’ll work on the PR to improve it, of course. I’ll start by reviewing the PR, figuring out where the 200+ lines came from, clean up the text according to your input, and ping you for a review later in the day.

Qq: how do I trigger the review app myself and get the link to it? Does it deploy attomatically, or should I do some magic comment on the pr?

Nate
(650) 933-7237

On Wed, Jul 1 2020 at 11:39 AM, Jorge Orpinel < notifications@github.com > wrote:

> 
> 
> 
> 
> 
> 
> 
> *@jorgeorpinel* commented on this pull request.
> 
> 
> 
> In content/docs/start/experiments.md (
> https://github.com/iterative/dvc.org/pull/1514#discussion_r448548570 ) :
> 
> >  And finally, we can compare `ROC` curves with a single command!
>  
> 
> ```dvc
> +$ dvc plots modify prc.json -x recall -y precision
> +$ git commit
> -am "Update plots format"
> +
>  $ dvc plots diff
> 
> file:///Users/dvc/example-get-started/plots.html
>  ```
> 
> 
> Well, we could just add -x and -y to the plots diff command, actually. See
> https://dvc.org/doc/command-reference/plots/diff#options
> 
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub (
> https://github.com/iterative/dvc.org/pull/1514#discussion_r448548570 ) , or
> unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AAAKU4M56DERMGNRXDZUC63RZN7EJANCNFSM4ONCDTDA
> ).
> 
> 
>

jorgeorpinel commented:
Sure @xnutsive take your time.

> Qq: how do I trigger the review app myself

Can't when the PR comes from a fork. We have to do it for you. But you can easily run the app locally! See https://dvc.org/doc/user-guide/contributing/docs

jorgeorpinel commented on content/docs/start/data-pipelines.md:
```diff
@@ -26,9 +26,17 @@ Get the sample code like this:
+    ├── prepare.py
+    ├── requirements.txt
+    └── train.py
+
+1 directory, 6 files
```
We can skip these 2 lines
```suggestion
```

jorgeorpinel replied:
I'll just commit this. (You'll have to `git pull` after this @xnutsive.)

xnutsive commented on content/docs/start/data-pipelines.md:
```diff
@@ -35,8 +35,6 @@ $ tree
     ├── prepare.py
     ├── requirements.txt
     └── train.py
-
```
👍

jorgeorpinel commented on content/docs/start/experiments.md:
```diff
@@ -221,3 +221,7 @@ All these commands also accept
 names) to compare. This is a powerful mechanism for navigating experiments to
 see the history, to pick the best ones, etc.
+
+If you report more than one metric to the `plots` file, you can provide 
+metric names to `dvc plots diff` command with `-x` and `-y` options.
+See more here: [`dvc plots diff` options](https://dvc.org/doc/command-reference/plots/diff#options).
```
Probably not needed. WDYT @shcheklein ?

If keeping this, I'd put it in a block quote (`>`) and start "The -x and -y options select which data series..." or something like that but it's going to be difficult to write it in a way that doesn't require further explanations (for example what is a data series?) so again, probably best to avoid this...

jorgeorpinel replied:
Maybe just "> See `dvc plots diff` for more info on its options, such as `-x` and `-y` used above."

jorgeorpinel replied:
Notice that there's no need to make manual links from DVC commands. they get auto-linked.

xnutsive replied:
The goal of this paragraph, when I wrote this, was to: 

1) Explain how to plot different metrics if needed, and explain that that's actually possible, spark interest to experiment with different metrics like that.
2) Give an example of how to do this (actually, the example plot above does this in an implicit way).

You guys probably have a better judgement on whether it's a good idea to explain this in the get started series.

jorgeorpinel replied:
I think we want to keep the GS as simple and focused as possible. Users with interest on this can click on the `dvc plots` link that already exists above and learn everything about metrics and plots. We also don't talk about `dvc plots show`, for example... No way to cover all these options in GS directly. I personally vote to remove this note altogether but I'm OK with a very short one like I suggested above, since this is the very bottom of the doc anyway.

shcheklein replied:
Yep, I would omit it as well. In other places if explain options we explain them in the `<details>` section. Otherwise PR looks great to me!

jorgeorpinel commented:
@xnutsive the Ci checks are failing. I think it's the formatting since Restyled is also failing. Please follow the formatting explained in the contribution guide 🙂

xnutsive commented:
@jorgeorpinel, that was my bad — hooks didn't run because I removed yarn, and I didn't pay attention. Fixed both the formatting, and the content. Thanks for being patient. ;)

shcheklein commented:
thanks @xnutsive !
